ios/browser/component_updater/ios_component_installer_policy.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/component_updater/ios_component_installer_policy.h"
+
+#include <sys/qos.h>
+#include <string>
+#include "crypto/sha2.h"
+
+#include "base/base64.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/json/json_string_value_serializer.h"
+#include "base/task/thread_pool.h"
+
+#include "brave/components/brave_component_updater/browser/brave_component.h"
+#include "components/component_updater/component_installer.h"
+#include "components/component_updater/component_updater_service.h"
+#include "components/crx_file/id_util.h"
+
+using brave_component_updater::BraveComponent;
+
+namespace {
+bool RewriteManifestFile(const base::FilePath& extension_root,
+                         const base::Value::Dict& manifest,
+                         const std::string& public_key) {
+  // Add the public key
+  DCHECK(!public_key.empty());
+
+  base::Value::Dict final_manifest = manifest.Clone();
+  final_manifest.Set("key", public_key);
+
+  std::string manifest_json;
+  JSONStringValueSerializer serializer(&manifest_json);
+  serializer.set_pretty_print(true);
+  if (!serializer.Serialize(final_manifest)) {
+    return false;
+  }
+
+  base::FilePath manifest_path =
+      extension_root.Append(FILE_PATH_LITERAL("manifest.json"));
+  if (!base::WriteFile(manifest_path, manifest_json)) {
+    return false;
+  }
+  return true;
+}
+
+std::string GetManifestString(base::Value::Dict* manifest,
+                              const std::string& public_key) {
+  manifest->Set("key", public_key);
+
+  std::string manifest_json;
+  JSONStringValueSerializer serializer(&manifest_json);
+  serializer.set_pretty_print(true);
+  if (!serializer.Serialize(*manifest)) {
+    return "";
+  }
+  return manifest_json;
+}
+}  // namespace
+
+namespace component_updater {
+// IOSComponentInstallerPolicy
+
+IOSComponentInstallerPolicy::IOSComponentInstallerPolicy(
```

Comment:
We definitely do not want to duplicate this stuff in ios. What is the reason you created these?

Reply:
Not much. This is my first major work on brave-core and I wasn't too comfortable moving the desktop file. Plus originally there were some differences in the iOS and desktop versions. But they ended up looking the same in the end. 

Looking at the desktop version of this file, it should be fairly easy to move to components and share it with iOS

Reply:
I'm using your components now

Reply:
@bridiver can we resolve this comment?